proto/messages.go

```diff
@@ -555,29 +755,41 @@ func ReadFetchReq(r io.Reader) (*FetchReq, error) {
 	return &req, nil
 }
 
-func (r *FetchReq) Bytes() ([]byte, error) {
+func (r *FetchReq) Bytes(version int16) ([]byte, error) {
 	var buf bytes.Buffer
 	enc := NewEncoder(&buf)
 
 	// message size - for now just placeholder
 	enc.Encode(int32(0))
 	enc.Encode(int16(FetchReqKind))
-	enc.Encode(int16(0))
+	enc.Encode(r.Version)
```

Comment:
I am a little bit confused about this part.  Why do we encode `r.Version` in message header,  but create body based on passed `version`? .  What will happen if `r.Version` and `version` is different?   Can server parse such messages correctly?  What are the cases when we need to call Bytes() with version different from version of request?

Reply:
It was added for the Response case, the Response structures currently don't have a version field but need to be encoded for a particular version according to the request. 

I'm happy to add a Version field to the Response structs to avoid this. I agree that it is not required for Requests and can even be confusing.

Reply:
Could you do it?  I think it would simplified code if we don't need to pass version in every `WriteTo` call.

Reply:
Hi Thomas.  It is a great PR and we would love to merge it. Do you need any help with this change?  Or do you prefer to merge it as it is?

Reply:
I'm a bit short on time this week. If someone has cycles to implement the requested changes then I'm totally fine with that.

Reply:
Hi Thomas.  I've fixed that part about version, but I am not sure how to add changes in this PR.  So I guess I'll merge this PR and create another one, with my fixes on top of it.  I'll be grateful if you'd take a look at it too.

Reply:
https://github.com/optiopay/kafka/pull/82